modules/web/server/src/main/scala/navigate/web/server/http4s/WebServerLauncher.scala

```diff
@@ -226,6 +228,21 @@ object WebServerLauncher extends IOApp with LogInitialization {
     case e: Exception       => Logger[F].error(e)("Navigate global error handler")
   }
 
+  /**
+   * Buffer log messages to be able to send old messages to new clients
+   */
+  def bufferLogMessages(log: Topic[IO, ILoggingEvent]): Resource[IO, Ref[IO, Seq[ILoggingEvent]]] =
+    val maxQueueSize = 30
+    for
+      buffer <- Ref.empty[IO, Seq[ILoggingEvent]].toResource
+      _      <- log
+                  .subscribe(1024)
+                  .evalMap(event => buffer.update(events => events.take(maxQueueSize - 1) :+ event))
```

Comment:
Shouldn't it be `events.takeRight(maxQueueSize - 1) :+ event` ? I think the proposed implementation keeps the same 29 messages and just replaces the last one.

Reply:
Yep, absolutely right. Great catch!